model/inline/src/main/scala/aqua/model/inline/state/Mangler.scala

```diff
@@ -1,65 +1,46 @@
 package aqua.model.inline.state
 
 import cats.data.State
+import aqua.mangler.ManglerState
 
 trait Mangler[S] {
   self =>
-  def getForbiddenNames: State[S, Set[String]]
-
-  def findNewNames(introduce: Set[String]): State[S, Map[String, String]]
-
-  def findNewName(introduce: String): State[S, String] =
-    findNewNames(Set(introduce)).map(_.getOrElse(introduce, introduce))
+  def getForbiddenNames: State[S, ManglerState]
 
   def findAndForbidName(introduce: String): State[S, String] =
-    for {
-      n <- findNewName(introduce)
-      _ <- forbid(Set(n))
-    } yield n
+    findAndForbidNames(Set(introduce)).map(_.getOrElse(introduce, introduce))
 
-  def findAndForbidNames(introduce: Set[String]): State[S, Map[String, String]] =
-    for {
-      n <- findNewNames(introduce)
-      _ <- forbid(introduce ++ n.values.toSet)
-    } yield n
+  def findAndForbidNames(introduce: Set[String]): State[S, Map[String, String]]
 
   def forbid(names: Set[String]): State[S, Unit]
 
-  def forbidName(name: String): State[S, Unit] =
-    forbid(Set(name))
-
   def transformS[R](f: R => S, g: (R, S) => R): Mangler[R] =
     new Mangler[R] {
 
-      val getForbiddenNames: State[R, Set[String]] =
+      val getForbiddenNames: State[R, ManglerState] =
         self.getForbiddenNames.transformS(f, g)
 
-      def findNewNames(introduce: Set[String]): State[R, Map[String, String]] =
-        self.findNewNames(introduce).transformS(f, g)
-
       def forbid(names: Set[String]): State[R, Unit] =
         self.forbid(names).transformS(f, g)
+
+      def findAndForbidNames(introduce: Set[String]): State[R, Map[String, String]] =
+        self.findAndForbidNames(introduce).transformS(f, g)
     }
 }
 
 object Mangler {
   def apply[S](implicit mangler: Mangler[S]): Mangler[S] = mangler
 
-  implicit object Simple extends Mangler[Set[String]] {
-    val getForbiddenNames: State[Set[String], Set[String]] = State.get
+  implicit object Simple extends Mangler[ManglerState] {
```

Comment:
```suggestion
  given Mangler[ManglerState] with {
```
and then `Mangler.Simple -> Mangler[ManglerState]`